addon/authorizers/token.js

```diff
@@ -73,27 +75,14 @@ export default Base.extend({
     ```
 
     @method authorize
-    @param {jqXHR} jqXHR The XHR request to authorize (see http://api.jquery.com/jQuery.ajax/#jqXHR)
+    @param {object} data
+    @param {function} block
   */
-  authorize: function(jqXHR) {
-    var token = this.buildToken();
+  authorize: function(data = {}, block = () => {}) {
+    const token = data[this.tokenPropertyName];
 
     if (this.get('session.isAuthenticated') && !Ember.isEmpty(token)) {
-      if(this.authorizationPrefix) {
-        token = this.authorizationPrefix + token;
-      }
-
-      jqXHR.setRequestHeader(this.authorizationHeaderName, token);
+      block(this.authorizationHeaderName, this.authorizationPrefix + token);
```

Comment:
Is it possible that this causes a regression when `authorizationPrefix` is set to `null`?

Using this branch and `authorizationPrefix: null` the authorization headers generated are `nullYekatoken..`.